lib/honeybadger.ex

```diff
@@ -257,10 +289,32 @@ defmodule Honeybadger do
       |> Map.put(:breadcrumbs, breadcrumbs)
 
     exception
-    |> Notice.new(metadata_with_breadcrumbs, stacktrace)
+    |> Notice.new(metadata_with_breadcrumbs, stacktrace, fingerprint)
+    |> put_notice_fingerprint()
     |> Client.send_notice()
   end
 
+  def notify(exception, metadata, stacktrace) when is_map(metadata) and is_list(stacktrace) do
+    Logger.warn("Reporting with notify/3 is deprecated, use notify/2 instead")
```

Comment:
Using `Logger.warn` is different than `IO.warn` as it isn't caught by the compiler or `--warnings-as-errors`. To properly deprecate this we'll need to use `IO.warn` instead.